CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a royal flush, false otherwise.</returns>
+        private static bool IsRoyalFlush(List<Card> cards)
+        {
+            return IsStraightFlush(cards) && cards[0].GetPokerValue() == 10;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a flush, false otherwise.</returns>
+        private static bool IsFlush(List<Card> cards)
+        {
+            Suit suit = cards[0].Suit;
+            foreach (Card card in cards)
+            {
+                if (card.Suit != suit)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a straight flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight flush, false otherwise.</returns>
+        private static bool IsStraightFlush(List<Card> cards)
+        {
+            return IsFlush(cards) && IsStraight(cards);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a four of a kind.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a four of a kind, false otherwise.</returns>
+        private static bool IsFourOfAKind(List<Card> cards)
+        {
+            return cards[0].SameVal(cards[3]);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a full house.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a full house, false otherwise.</returns>
+        private static bool IsFullHouse(List<Card> cards)
+        {
+            if (!cards[0].SameVal(cards[1]) || !cards[3].SameVal(cards[4]))
+            {
+                return false;
+            }
+
+            if (cards[2].SameVal(cards[1]) || cards[2].SameVal(cards[3]))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a straight.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight, false otherwise.</returns>
+        private static bool IsStraight(List<Card> cards)
+        {
+            // Special case: 2345A
+            if (cards[0].GetPokerValue() == 2 && cards[1].GetPokerValue() == 3 && cards[2].GetPokerValue() == 4 && cards[3].GetPokerValue() == 5 && cards[4].IsAce())
+            {
+                return true;
+            }
+
+            for (int i = 1; i < cards.Count; i++)
```

Comment:
This can be simplified using Linq:

First, get the expected sum based on the value of `cards[0]`. If `cards[0]` is 5, then the expected sum would be 5+6+7+8+9 = 35.
Second, get the sum of the cards: `cards.Sum(card => card.GetPokerValue());`
Third, return if they are equal or not.

This would save a lot of resources on indexing the list a bunch of times.